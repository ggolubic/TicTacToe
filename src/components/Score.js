import React from "react";
import styled from "styled-components";
import Player from "./Player";

export const Score = ({ score }) => {
  return (
    <ScoreContainer>
      <Player num={1} score={score.x} />
      <Player num={2} score={score.o} />
      <h2>Draw: {score.draw}</h2>
    </ScoreContainer>
  );
};

const ScoreContainer = styled.div`
  font-family: "Montserrat", sans-serif;
  width: 300px;
  margin: 0 auto;
  text-align: center;
  margin-top: 50px;
`;
