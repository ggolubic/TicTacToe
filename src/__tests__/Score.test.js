import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import Score from "../components/Score";

describe("Score component", () => {
  it("Should render with default props", () => {
    const component = renderer.create(<Score />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it("Should render with passed props", () => {
    const component = renderer.create(
      <Score score={{ x: 1, o: 1, draw: 0 }} />
    );
    expect(component).toMatchSnapshot();
  });
});
