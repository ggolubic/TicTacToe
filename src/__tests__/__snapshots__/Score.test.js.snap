// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score component Should render with default props 1`] = `
.c1 {
  color: #545454;
}

.c2 {
  color: #ffd02d;
}

.c0 {
  font-family: "Montserrat",sans-serif;
  width: 300px;
  margin: 0 auto;
  text-align: center;
  margin-top: 50px;
}

<div
  className="c0"
>
  <h2
    className="c1"
  >
    Player 
    1
    : 
    0
  </h2>
  <h2
    className="c2"
  >
    Player 
    2
    : 
    0
  </h2>
  <h2>
    Draw: 
    0
  </h2>
</div>
`;

exports[`Score component Should render with passed props 1`] = `
.c1 {
  color: #545454;
}

.c2 {
  color: #ffd02d;
}

.c0 {
  font-family: "Montserrat",sans-serif;
  width: 300px;
  margin: 0 auto;
  text-align: center;
  margin-top: 50px;
}

<div
  className="c0"
>
  <h2
    className="c1"
  >
    Player 
    1
    : 
    1
  </h2>
  <h2
    className="c2"
  >
    Player 
    2
    : 
    1
  </h2>
  <h2>
    Draw: 
    0
  </h2>
</div>
`;
